airbyte-integrations/connectors/source-freshdesk/source_freshdesk/streams.py

```diff
@@ -0,0 +1,273 @@
+        # with default value 30 days look back.
+        self.start_date = pendulum.parse(start_date) if start_date else pendulum.now() - pendulum.duration(days=30)
+    
+    @property
+    def url_base(self) -> str:
+        return f"https://{self.domain.rstrip('/')}/api/v2"
+    
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        if response.status_code == requests.codes.too_many_requests:
+            return float(response.headers.get("Retry-After"))
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        return {
+            "Content-Type": "application/json",
+            "User-Agent": "Airbyte",
+        }
+    
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        try:
+            link_header = response.headers.get("Link")
+            if not link_header:
+                return {}
+            match = LINK_REGEX.search(link_header)
+            next_url = match.group(1)
+            params = parse.parse_qs(parse.urlparse(next_url).query)
+            return self.parse_link_params(link_query_params=params)
+        except Exception as e:
+            raise KeyError(f"error parsing next_page token: {e}")
+    
+    def parse_link_params(self, link_query_params: Mapping[str, List[str]]) -> Mapping[str, Any]:
+        return {"per_page": link_query_params['per_page'][0], "page": link_query_params['page'][0]}
+    
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {"per_page": self.result_return_limit}
+        if next_page_token and "page" in next_page_token:
+            params["page"] = next_page_token["page"]
+        return params
+    
+    def _consume_credit(self, credit):
+        """Consume call credit, if there is no credit left within current window will sleep til next period"""
+        if self._call_credit:
+            self._call_credit.consume(credit)
+    
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        self._consume_credit(self.call_credit)
+        yield from super().read_records(sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state)
+    
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        data = response.json()
+        if not data:
+            return
+        for element in data:
+            yield element
+
+
+class IncrementalFreshdeskStream(FreshdeskStream, ABC):
```

Comment:
```suggestion
class IncrementalFreshdeskStream(FreshdeskStream, IncrementalMixin):
```
The latest CDK introduces the `IncrementalMixin` class. I suggest you use it and set the state value in `read_records` as suggested [here](https://docs.airbyte.com/connector-development/cdk-python/incremental-stream/#incrementalmixin). `get_updated_state` is now deprecated.

Reply:
Added in [abcae41](https://github.com/airbytehq/airbyte/pull/12334/commits/abcae4135576a3541a76979a09babf3a955c6b8f)